### PR TITLE
[TASK] Add support for issueing PKI Certificates

### PR DESCRIPTION
**Pull request by UXabre**
I've added the ability to issue certificates when using a PKI secrets engine.
I've purposely kept the options undefined, as the PKI is a very versatile engine. The only thing that might bother me, is the path, being "<pki_engine_name>/issue/<pki_role>" is to be used to issue a request

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,13 +39,23 @@ As result you should get secret data, e.g.
...
+The common_name and ttl are optional, but there are [more options to configure](https://www.vaultproject.io/api/secret/pki/index.html#generate-certificate)
+
 ## Contributing
 
 Fork and send a Pull Request. Thanks!
 
 ## Copyright
 
-Copyright (c) 2018 dmTECH GmbH, [dmtech.de](https://www.dmtech.de/)
+Copyright (c) 2020 dmTECH GmbH, [dmtech.de](https://www.dmtech.de/)
```
Comment (ekohl) [inline README.md]: Shouldn't this be 2018-2020?

Comment (UXabre) [reply README.md]: Oh yes, sorry about that!

File: README.md [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (timogoebel) [inline README.md]: Nit: can we remove this empty line?

File: README.md [hunk 3]
```diff
@@ -33,13 +39,23 @@ As result you should get secret data, e.g.
 {:foo=>"bar"}
 ```
 
+### vault_issue_certificate(vault_connection_name, pki_role_path, options...)
+Issueing certificates is just as easy. Be sure to have a correctly set-up PKI, meaning, configure it so you can generate certificates from within the Vault UI. This means that you'll have had to set-up a CA or Intermediate CA. Once done, you can generate a certificate like this:
+
+
+```
+<%= issue_certificate('MyVault', 'pkiEngine/issue/testRole', common_name: 'test.mydomain.com', ttl: '10s') %>
```
Comment (kamils-iRonin) [inline README.md]: `vault_issue_certificate` is the correct macro name, right?

File: README.md (hunk 2 above)
Comment (UXabre) [reply README.md]: Yes, I'll remove it (sorry, usually I notice these things but I overlooked it this time)

**Comment by UXabre** [comment]
I'll add tests latter tonight! Thanks for reviewing so far!

File: README.md (hunk 3 above)
Comment (UXabre) [reply README.md]: Corrected it! Thanks for spotting this!

**Comment by UXabre** [comment]
I've added a test, as well as adding the macro to the safe functions of the template function

thanks @kamils-iRonin, @timogoebel & @ekohl for the reviews so far :-)

File: README.md (hunk 2 above)
Comment (timogoebel) [reply README.md]: This line still wants to be removed.

**Comment by UXabre** [comment]
Hi @timogoebel ,

I overlooked the white line :-( sorry! But it's resolved now! Thanks for your review!

**Comment by timogoebel** [comment]
@UXabre: Looks great, thanks! I'll see to it that we fix the CI pipeline here. Then we can rebase this PR and merge it when tests pass.

**Comment by timogoebel** [comment]
@UXabre: Do you mind rebasing this?

**Comment by UXabre** [comment]
Hi @timogoebel ,

Rebased and ready to go! :-) Thanks for fixing the test suite! I noticed some rubocop issues and fixed them as well.

Thanks for and happy reviewing!

**Comment by timogoebel** [comment]
Thanks, @UXabre.

**Comment by timogoebel** [comment]
@UXabre: I've just released version 0.1.0 of this gem that contains these changes!

**Pull request merged**